### DISCUSSION TITLE
Why isn't fetchPartials exposed for ajax requests?

### DISCUSSION THREAD

**Discussion by debel** [freshframework/fresh#2487] [Q&A] [answered]
Hello,

I've experimenting with different technologies and very much like the minimalistic and yet powerful setup of Deno+ Fresh. Big thanks to everyone involved in building this. Great job!

So my question is: Why isn't fetchPartials exposed for ajax requests?

Context: Looking at the network requests for the client-side navigation I noticed that the requests are coming from the `fetchPartials` functions in the client's `main.ts` and I though it might be a great utility to allow the implementer to manually call this function to fetch pre-rendered data.

Am I thinking wrongly about this? Did anyone else wander how to just make a, say, POST request that instantly updates the DOM (like magic). Is there a reason for not exposing it? It can be restricted to only allow fetching same-origin and known routes (as per manifest).

WDYT, looking forward to reading your thoughts :)

**Comment by marvinhagemeister** [comment, marked as answer]
Main reason we haven't exposed it yet is that we wanted to wait a little until the inner internal APIs settle, so that we are confident in making apps rely on it. So far it hasn't changed much, so that's a good sign that we're moving in the right direction. I'm hoping to go through the partials-related bugs in the issue tracker and kinda see where we end up with and if it stays the same. If it does, then it would feel like a good time to expose it.

**Reply by debel** [reply]
Thanks for the response, and again, thanks for all your efforts. Glad to hear that it's going to be exposed - I think it would be a great addition to the toolbox and save on needing to handle all the request state client side. Also kind of glad to hear I'm not doing things totally backwards 😄

**Reply by Sleepful** [reply]
Looking forward to this! It seems to be a requirement in order to have the best experience for content that needs to be loaded after the page has rendered for the user, and without manual intervention from the user (e.g. not through a click of `a`, `button` or `form`)
